src/prefect/storage/docker.py

```diff
@@ -380,6 +380,7 @@ def _build_image(self, push: bool = True) -> tuple:
                 path="." if self.dockerfile else tempdir,
                 dockerfile=dockerfile_path,
                 tag="{}:{}".format(full_name, self.image_tag),
+                rm=self.build_kwargs.pop("rm", True),
                 **self.build_kwargs,
```

Comment:
If `build` is called more than once and the user sets `rm=False`, this will not behave as intended.

Reply:
There isn't really a use-case that I know of, but people use things in weird ways and it'd be a sneaky bug to sort out later :)

Reply:
They could be sharing `build_kwargs` across multiple flows that they are registering. We don't copy it in `__init__` so we shouldn't mutate the input value. If you move the logic to the constructor, make sure you create a copy of the input `build_kwargs`.

Reply:
Thanks, that's a fair example! 
I made the changes :) I ended up using copy (and not deepcopy) because I thought it would be unlikely that the `__init__` would need to mutate the values of `build_kwargs` passed in by the user. Let me know if you think `deepcopy` makes more sense in this context!

Reply:
Just `copy` makes sense!